sql-plugin/src/main/scala/com/nvidia/spark/rapids/Arm.scala

```diff
@@ -31,6 +31,15 @@ trait Arm {
     }
   }
 
+  /** Executes the provided code block and then closes the Option[resource] */
+  def withResource[T <: AutoCloseable, V](r: Option[T])(block: Option[T] => V): V = {
+    try {
+      block(r)
+    } finally {
+      r.map(_.close())
```

Comment:
nit: technically this should be foreach instead of map. very minor.

Reply:
Done